lib/pure/random.nim

```diff
@@ -254,9 +255,7 @@ proc rand*(r: var Rand; max: range[0.0 .. high(float)]): float {.benign.} =
     result = (cast[float](u) - 1.0) * max
 
 proc rand*[T: Ordinal or SomeFloat](max: T): T {.benign.} =
-  ## Returns a random floating point number in the range `T(0) .. max`.
-  ## 
-  ## Allowed types for `T` are integers, floats, and enums without holes.
+  ## Returns a random number in the range `T(0) .. max`.
```

Comment:
add:
```
`max` must be >= T(0).
```

in doc comment and add `assert max >= T(0)` in code otherwise you get bad error messages


```nim
  import std/random
  for i in 0..<100:
    echo rand(-2i8)
```
fails with:
```
random.nim(285)
 -2 notin 0 .. 9223372036854775807 [RangeDefect]
```

Reply:
But `0 .. -1` is not correct too.

Reply:
The issue exists before.

Reply:
```nim
import random


echo rand(-1)
echo rand(12 .. -12)

var r = initRand(12)

echo r.rand(12 .. -12)
```
https://play.nim-lang.org/#ix=2H7X